lib/plugins.js

```diff
@@ -30,18 +33,35 @@ function mergeByName(acceptor, donor, name, onConflict) {
   }, this);
 }
 
-var plugins = [],
-    dirlist;
+var cachedPlugins = {};
 
-try {
-  dirlist = fs.readdirSync(CWD);
-} catch (err) {
-  dirlist = [];
+function _setMockPlugins(pluginPath, plugins) {
+  cachedPlugins[pluginPath] = plugins;
 }
-  
-dirlist.reduce(function(plugins, subdir) {
-    var _path = path.resolve(CWD, subdir, 'manifest.json'),
-        manifest;
+
+function getPlugins(config, ignoreCache) {
```

Comment:
Yes, I understand how it can be used, but I don't see where we use this logic.

I'm not sure that reinitialisation of plugins on the fly is good think. It's useful in plugins development, but for end user this can cause some unexpected problems. Can we avoid caching this logic on firs step?